jetson/science_bridge/src/__main__.py

```diff
@@ -0,0 +1,391 @@
+                                 enable=int(struct.enabled))
+
+        while(len(message) < 30):
+            # Add an extra 1 for padding
+            message += "1"
+        print(message)
+        self.ser.close()
+        self.ser.open()
+        if self.ser.isOpen():
+            self.ser.write(bytes(message, encoding='utf-8'))
+        pass
+
+    def heater_auto_transmit(self, channel, msg):
+        # Send the nucleo a message telling if auto shutoff for heaters is off or on
+        struct = HeaterAutoShutdown.decode(msg)
+        message = "$AutoShutoff,{enable},1111"
+        message = message.format(enable=int(struct.auto_shutdown_enabled))
+        while(len(message) < 30):
+            # Add an extra 1 for padding
+            message += "1"
+        print(message)
+        self.ser.close()
+        self.ser.open()
+        if self.ser.isOpen():
+            self.ser.write(bytes(message, encoding='utf-8'))
+        pass
+
+    def mosfet_transmit(self, channel, msg):
+        print("Mosfet callback")
+        struct = MosfetCmd.decode(msg)
+
+        message = "$Mosfet,{device},{enable},1111111"
+        translated_device = struct.device
+        # Translate individual pins to their respective nucleo pin
+        # Laser and UV LED share pins 1 and 2
+        # const int8_t rLed = 0;
+        # const int8_t gLed = 1;
+        # const int8_t bLed = 2;
+        # const int8_t Laser = 3;
+        # const int8_t UVLED = 4;
+        # const int8_t whiteLED = 5;
+        # const int8_t uvBulb = 6;
+        # const int8_t nichWire0 = 7;
+        # const int8_t nichWire1 = 8;
+        # const int8_t nichWire2 = 9;
+        # const int8_t ramanLaser = 10;
+        # Starts from 0-2 for all the leds
+        # Resets to 1 starting at Laser(3) so offset by 2
+        # if (translated_device > 2):
+        #     translated_device -= 2
+        message = message.format(device=translated_device,
+                                 enable=int(struct.enable))
+        print(message)
+        while(len(message) < 30):
+            message += ","
+        self.ser.close()
+        self.ser.open()
+        if self.ser.isOpen():
+            self.ser.write(bytes(message, encoding='utf-8'))
+        print("Mosfet Received")
+        pass
+
+    def nav_status(self, channel, msg):  # TODO - fix make better
+        print("Received nav req")
+        # Off, Done, Else
+
+        struct = NavStatus.decode(msg)
+        message = "$Mosfet,{device},{enable},11111111"  # TODO - PROBABLY OUTDATED, WRONG PADDING
+        # All Leds are 1 digit so hardcode in padding
+
+        # Off = Blue
+        if struct.nav_state_name == "Off":
+            print("navstatus off")
+            offmessage = message.format(device=2, enable=1) + "1"
+            self.ser.write(bytes(offmessage, encoding='utf-8'))
+            prev = 2
+        # Done = Flashing green
+        elif struct.nav_state_name == "Done":
+            print("navstatus Done")
```

Comment:
i think the blinking can really be done the nucleo end..here just have it send blink once (i.e when the status changes to done) and on the nucleo until it gets a state change it can be changing the mosfet in the loop

Reply:
also are you doing to have a #define AUTON in the nucleo code.....?

Reply:
Yea the Auton led code right now has not been touched or planned for yet... and yes there is much room for improvement....

Reply:
ok....ideally lets test it tonight so they have LEDs tmrw, & hopefully it's worked on this week/next